src/main/webapp/script.js

```diff
@@ -80,16 +80,17 @@ function loadSearchResults() {
       .then((data) => {
         const parsedData = JSON.parse(data);
         schoolsFetchedDataList = parsedData['results'];
-        localStorage.setItem('schoolsFetchedDataList',
-            JSON.stringify(schoolsFetchedDataList));
+
         const schoolsDataList = [];
+        const schoolsFetchedDataById = {};
```

Comment:
this is awesome. Can we add some comments on its data structure definition like `// Holds schoolId => schoolInfo in a data dictionary for O(1) retrieval`, or rename the variable as it is like `schoolIdToInfoDataDictionary`. Otherwise, the name is a little bit confused.